sled-agent/src/bin/sled-agent.rs

```diff
@@ -61,10 +61,13 @@ async fn do_run() -> Result<(), CmdError> {
                 rss_config_path
             };
             let rss_config = if rss_config_path.exists() {
-                Some(
+                let mut rss_config =
                     RackInitializeRequest::from_file(rss_config_path)
-                        .map_err(|e| CmdError::Failure(anyhow!(e)))?,
-                )
+                        .map_err(|e| CmdError::Failure(anyhow!(e)))?;
+                if rss_config.skip_timesync.is_none() {
+                    rss_config.skip_timesync = config.skip_timesync;
```

Comment:
We chatted about this but just noting it - I think the overall changes on this PR to make RSS respect `skip_timesync` are right, but I don't think we should add an `rss_config.skip_timesync` field. In this file, we already have access to `config.skip_timesync` from the sled-agent config file; maybe we add that as an argument to `start_rack_initialize()` down on line 81?

Reply:
I was just looking at that. That function delegates to an HTTP end point, but maybe I'm over-rotating on that; I'm going to poke at it tomorrow morning and see if I can get it to go.

Reply:
I think this is ready to go now. Plumbing the boolean through via an additional argument kicked over Clippy's complaint threshold on the number of arguments to a function. Frankly, Clippy was right. Anyway, I've bundled both the request and the `skip_timesync` bool into a small struct and modified things to pass that, which I'm happier with.